dm/pkg/checker/privilege.go

```diff
@@ -196,15 +234,37 @@ func verifyPrivileges(result *Result, grants []string, expectedGrants map[mysql.
 	}
 
 	if len(lackGrants) != 0 {
-		lackGrantsStr := make([]string, 0, len(lackGrants))
-		for g := range lackGrants {
-			lackGrantsStr = append(lackGrantsStr, mysql.Priv2Str[g])
+		var buffer bytes.Buffer
+		// generate error message, for example
+		// lack of privilege1: {tableID1, tableID2, ...};lack of privilege2...
+		for g, tableMap := range lackGrants {
+			buffer.WriteString("lack of ")
+			buffer.WriteString(mysql.Priv2Str[g])
+			buffer.WriteString(" privilege")
+			i := 0
+			for schema, tables := range tableMap {
```

Comment:
please add an unit test for this message

Reply:
I mean I want to check if the full message is well formatted, `lack of .* privilege.*;` is not that confident to me.

Reply:
added more detailed errMsg test in https://github.com/pingcap/tiflow/pull/3894/commits/d171491a638e1db3c8c1aafd17e1ed11422bb860. ptal